lib/OpenQA/Task/Job/Restart.pm

```diff
@@ -17,6 +18,13 @@ sub restart_delay { $ENV{OPENQA_JOB_RESTART_DELAY} // 5 }
 sub restart_openqa_job ($minion_job, $openqa_job) {
     my $cloned_job_or_error = $openqa_job->auto_duplicate;
     my $is_ok = ref $cloned_job_or_error || $cloned_job_or_error =~ qr/(already.*clone|direct parent)/i;
+    if ($is_ok) {
+        my $openqa_job_id = $openqa_job->id;
+        my $result = ref $cloned_job_or_error ? $cloned_job_or_error->{cluster_cloned} : $cloned_job_or_error;
+        my %event_data = (id => $openqa_job_id, result => $result, auto => 1);
```

Comment:
We shouldn't emit an event if cloning failed.
```suggestion
    if (ref $cloned_job_or_error) {
        my %event_data = (id => $openqa_job->id, result => $cloned_job_or_error->{cluster_cloned}, auto => 1);

```

Reply:
Sorry, to be more clear: I meant "when cloning actually happened".
```
my $is_ok = ref $cloned_job_or_error || $cloned_job_or_error =~ qr/(already.*clone|direct parent)/i;
```
In the latter case coning did not happen.